src/Fixer/Semicolon/MultilineWhitespaceBeforeSemicolonsFixer.php

```diff
@@ -0,0 +1,292 @@
+
+            // appended new line to the last method call
+            $newline = new Token([T_WHITESPACE, $lineEnding.$indent]);
+
+            // insert the new line with indented semicolon
+            $tokens->insertAt($index, [$newline, new Token(';')]);
+        }
+    }
+
+    /**
+     * Find the index for the new line. Return the given index when there's no new line.
+     *
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return int
+     */
+    private function getNewLineIndex($index, Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        for ($index, $count = count($tokens); $index < $count; ++$index) {
+            if (false !== strstr($tokens[$index]->getContent(), $lineEnding)) {
+                return $index;
+            }
+        }
+
+        return $index;
+    }
+
+    /**
+     * Find the indentation of the chained call at $index.
+     *
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return null|string
+     */
+    private function getIndentation($index, Tokens $tokens)
+    {
+        // semicolon followed by a closing bracket?
+        if (!$tokens[$index]->equals(')')) {
+            return null;
+        }
+
+        // find opening bracket
+        $openingBrackets = 1;
+        for (--$index; $index > 0; --$index) {
+            if ($tokens[$index]->equals(')')) {
+                ++$openingBrackets;
+
+                continue;
+            }
+
+            if ($tokens[$index]->equals('(')) {
+                if (1 === $openingBrackets) {
+                    break;
+                }
+                --$openingBrackets;
+            }
+        }
+
+        // method name
+        if (!$tokens[--$index]->isGivenKind(T_STRING)) {
+            return null;
+        }
+
+        // ->
+        if (!$tokens[--$index]->isGivenKind(T_OBJECT_OPERATOR)) {
+            return null;
+        }
+
+        // white space
+        if (!$tokens[--$index]->isGivenKind(T_WHITESPACE)) {
+            return null;
+        }
+
+        $closingBrackets = 0;
+        for ($index; $index >= 0; --$index) {
+            if ($tokens[$index]->equals(')')) {
+                ++$closingBrackets;
+            }
+
+            if ($tokens[$index]->equals('(')) {
+                --$closingBrackets;
+            }
+
+            // must be the variable of the first call in the chain
+            if ($tokens[$index]->isGivenKind(T_VARIABLE) && 0 === $closingBrackets) {
+                if ($tokens[--$index]->isGivenKind(T_WHITESPACE)) {
+                    return $this->getIndentAt($tokens, $index);
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index  index of the indentation token
+     *
+     * @return null|string
+     */
+    private function getIndentAt(Tokens $tokens, $index)
```

Comment:
How is this method different to `getIndentation()`?

The have the same arguments just bit differt yet synonymous name. Using more specific name would be better